presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/ThriftExceptions.java

```diff
@@ -47,6 +48,6 @@ public static PrestoException toPrestoException(Exception e)
 
     public static <T> ListenableFuture<T> catchingThriftException(ListenableFuture<T> future)
     {
-        return catchingAsync(future, Exception.class, e -> immediateFailedFuture(toPrestoException(e)));
+        return catchingAsync(future, Exception.class, e -> immediateFailedFuture(toPrestoException(e)), directExecutor());
```

Comment:
> `Remove deprecated usage of catchingAsync`

This is more like "fix" rather than "remove"

Reply:
Agreed, I'll update the commit message